LOOKUP does not handle Array form correctly

**Description**

There are two ways to use LOOKUP: Vector form and Array form. For more info, see https://support.microsoft.com/en-us/office/lookup-function-446d94af-663b-451d-8251-369d5e3864cb

Excelize seems to support both, but the Array form does not work as expected. It returns the lookup value instead of the result

**Steps to reproduce the issue:**
1. Add these two tests to calc_test.go, they should be equivalent and return the same result:
```go
	mathCalc := map[string]string{
                ...
		"=LOOKUP(E3,E2:E5,F2:F5)":      "22100",
		"=LOOKUP(E3,E2:F5)":            "22100",
```
I've done this in my lookup-array-from branch: https://github.com/stanim/excelize/blob/lookup-array-form/calc_test.go#L1135

2. Run `go test`.

**Describe the results you received:**

```
> go test
--- FAIL: TestCalcCellValue (7.87s)
    calc_test.go:1218: 
        	Error Trace:	calc_test.go:1218
        	Error:      	Not equal: 
        	            	expected: "22100"
        	            	actual  : "North 2"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-22100
        	            	+North 2
        	Test:       	TestCalcCellValue
        	Messages:   	=LOOKUP(E3,E2:F5)
```

**Describe the results you expected:**

I expected the tests to pass.

**Output of `go version`:**

```text
go version go1.16.7 linux/amd64
```

**Excelize version or commit ID:**

```text
> git log
commit a55f354eb3d0c6c1b9a543ff8ff98227aa6063a6 (HEAD -> lookup, origin/master, origin/HEAD, master)
Author: xuri <xuri.me@gmail.com>
Date:   Tue Aug 17 00:01:44 2021 +0800

    This closes #989, closes #990
    
    New API: `SetRowStyle` support for set style for the rows
    Update documentation for the `GetRows`, `SetCellStyle` and `SetColStyle`
```

**Environment details (OS, Microsoft Excel™ version, physical, etc.):**

- OS: Archlinux 5.13.10-arch1-1
- WPS Spreadsheets (version wps-office 11.1.0.10702-1)
- LibreCalc (version libreoffice-fresh 7.1.5-2)

--- a/calc.go
+++ b/calc.go
@@ -7263,7 +7263,11 @@ func (fn *formulaFuncs) LOOKUP(argsList *list.List) formulaArg {
 	if lookupVector.Type != ArgMatrix && lookupVector.Type != ArgList {
 		return newErrorFormulaArg(formulaErrorVALUE, "LOOKUP requires second argument of table array")
 	}
-	cols, matchIdx := lookupCol(lookupVector), -1
+	arrayForm := lookupVector.Type == ArgMatrix
+	if arrayForm && len(lookupVector.Matrix) == 0 {
+		return newErrorFormulaArg(formulaErrorVALUE, "LOOKUP requires not empty range as second argument")
+	}
+	cols, matchIdx := lookupCol(lookupVector, 0), -1
 	for idx, col := range cols {
 		lhs := lookupValue
 		switch col.Type {
@@ -7280,9 +7284,13 @@ func (fn *formulaFuncs) LOOKUP(argsList *list.List) formulaArg {
 			break
 		}
 	}
-	column := cols
+	var column []formulaArg
 	if argsList.Len() == 3 {
-		column = lookupCol(argsList.Back().Value.(formulaArg))
+		column = lookupCol(argsList.Back().Value.(formulaArg), 0)
+	} else if arrayForm && len(lookupVector.Matrix[0]) > 1 {
+		column = lookupCol(lookupVector, 1)
+	} else {
+		column = cols
 	}
 	if matchIdx < 0 || matchIdx >= len(column) {
 		return newErrorFormulaArg(formulaErrorNA, "LOOKUP no result found")
@@ -7291,13 +7299,13 @@ func (fn *formulaFuncs) LOOKUP(argsList *list.List) formulaArg {
 }
 
 // lookupCol extract columns for LOOKUP.
-func lookupCol(arr formulaArg) []formulaArg {
+func lookupCol(arr formulaArg, idx int) []formulaArg {
 	col := arr.List
 	if arr.Type == ArgMatrix {
 		col = nil
 		for _, r := range arr.Matrix {
 			if len(r) > 0 {
-				col = append(col, r[0])
+				col = append(col, r[idx])
 				continue
 			}
 			col = append(col, newEmptyFormulaArg())

--- a/calc_test.go
+++ b/calc_test.go
@@ -1133,6 +1133,7 @@ func TestCalcCellValue(t *testing.T) {
 		"=LOOKUP(F8,F8:F9,D8:D9)":      "Feb",
 		"=LOOKUP(E3,E2:E5,F2:F5)":      "22100",
 		"=LOOKUP(E3,E2:F5)":            "22100",
+		"=LOOKUP(1,MUNIT(1))":          "1",
 		"=LOOKUP(1,MUNIT(1),MUNIT(1))": "1",
 		// ROW
 		"=ROW()":                "1",
@@ -2092,6 +2093,7 @@ func TestCalcCellValue(t *testing.T) {
 		"=LOOKUP()":                     "LOOKUP requires at least 2 arguments",
 		"=LOOKUP(D2,D1,D2)":             "LOOKUP requires second argument of table array",
 		"=LOOKUP(D2,D1,D2,FALSE)":       "LOOKUP requires at most 3 arguments",
+		"=LOOKUP(1,MUNIT(0))":           "LOOKUP requires not empty range as second argument",
 		"=LOOKUP(D1,MUNIT(1),MUNIT(1))": "LOOKUP no result found",
 		// ROW
 		"=ROW(1,2)":          "ROW requires at most 1 argument",

--- a/calc_test.go
+++ b/calc_test.go
@@ -1131,6 +1131,8 @@ func TestCalcCellValue(t *testing.T) {
 		// LOOKUP
 		"=LOOKUP(F8,F8:F9,F8:F9)":      "32080",
 		"=LOOKUP(F8,F8:F9,D8:D9)":      "Feb",
+		"=LOOKUP(E3,E2:E5,F2:F5)":      "22100",
+		"=LOOKUP(E3,E2:F5)":            "22100",
 		"=LOOKUP(1,MUNIT(1),MUNIT(1))": "1",
 		// ROW
 		"=ROW()":                "1",

